test/controllers/responsiblepeople/address/AdditionalAddressUKControllerSpec.scala

```diff
@@ -0,0 +1,326 @@
+        val additionalAddress = ResponsiblePersonAddress(UKAddress, Some(ZeroToFiveMonths))
+        val history = ResponsiblePersonAddressHistory(additionalAddress = Some(additionalAddress))
+        val responsiblePeople = ResponsiblePerson(personName = personName, addressHistory = Some(history))
+
+        when(additionalAddressUKController.dataCacheConnector.fetch[Seq[ResponsiblePerson]](any(), any())
+          (any(), any())).thenReturn(Future.successful(Some(Seq(responsiblePeople))))
+
+        val result = additionalAddressUKController.get(RecordId)(request)
+        status(result) must be(OK)
+
+        val document = Jsoup.parse(contentAsString(result))
+        document.title must be(pageTitle)
+        document.select("input[name=addressLine1]").`val` must be("Line 1")
+        document.select("input[name=addressLine2]").`val` must be("Line 2")
+        document.select("input[name=addressLine3]").`val` must be("Line 3")
+        document.select("input[name=addressLine4]").`val` must be("")
+        document.select("input[name=postcode]").`val` must be("AA1 1AA")
+      }
+
+      "display 404 NotFound" when {
+
+        "person cannot be found" in new Fixture {
+
+          val responsiblePeople = ResponsiblePerson()
+
+          when(additionalAddressUKController.dataCacheConnector.fetch[Seq[ResponsiblePerson]](any(), any())(any(), any()))
+            .thenReturn(Future.successful(Some(Seq(responsiblePeople))))
+
+          val result = additionalAddressUKController.get(RecordId)(request)
+          status(result) must be(NOT_FOUND)
+        }
+
+      }
+    }
+
+    "post is called" must {
+      "respond with SEE_OTHER" when {
+
+        "all the mandatory UK parameters are supplied" in new Fixture {
+
+          val requestWithParams = request.withFormUrlEncodedBody(
+            "isUK" -> "true",
+            "addressLine1" -> "Line 1",
+            "addressLine2" -> "Line 2",
+            "postCode" -> "AA1 1AA"
+          )
+          val UKAddress = PersonAddressUK("Line 1", "Line 2", Some("Line 3"), None, "AA1 1AA")
+          val additionalAddress = ResponsiblePersonAddress(UKAddress, Some(ZeroToFiveMonths))
+          val history = ResponsiblePersonAddressHistory(additionalAddress = Some(additionalAddress))
+          val responsiblePeople = ResponsiblePerson(addressHistory = Some(history))
+
+          when(additionalAddressUKController.dataCacheConnector.fetch[Seq[ResponsiblePerson]](any(), any())(any(), any()))
+            .thenReturn(Future.successful(Some(Seq(responsiblePeople))))
+          when(additionalAddressUKController.dataCacheConnector.save[PersonName](any(), any(), any())(any(), any()))
+            .thenReturn(Future.successful(emptyCache))
+
+          val result = additionalAddressUKController.post(RecordId)(requestWithParams)
+
+          status(result) must be(SEE_OTHER)
+
+//          val captor = ArgumentCaptor.forClass(classOf[DataEvent])
```

Comment:
If we do not need this can we remove it please?

Reply:
done